app/src/main/java/org/oppia/app/home/HomeFragmentPresenter.kt

```diff
@@ -21,35 +31,68 @@ private const val EXPLORATION_ID = TEST_EXPLORATION_ID_5
 /** The controller for [HomeFragment]. */
 @FragmentScope
 class HomeFragmentPresenter @Inject constructor(
-  activity: AppCompatActivity,
+  private val activity: AppCompatActivity,
   private val fragment: Fragment,
   private val viewModelProvider: ViewModelProvider<UserAppHistoryViewModel>,
+  private val promotedStoryViewModel: ViewModelProvider<PromotedStoryViewModel>,
   private val userAppHistoryController: UserAppHistoryController,
   private val explorationDataController: ExplorationDataController,
+  private val topicListController: TopicListController,
   private val logger: Logger
 ) {
-
   private val routeToExplorationListener = activity as RouteToExplorationListener
+  private val routeToTopicListener = activity as RouteToTopicListener
+
+  private val itemList: MutableList<Any> = ArrayList()
+
+  private lateinit var topicListAdapter: TopicListAdapter
+
+  private lateinit var binding: HomeFragmentBinding
 
   fun handleCreateView(inflater: LayoutInflater, container: ViewGroup?): View? {
-    val binding = HomeFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
+    binding = HomeFragmentBinding.inflate(inflater, container, /* attachToRoot= */ false)
     // NB: Both the view model and lifecycle owner must be set in order to correctly bind LiveData elements to
     // data-bound view models.
+
+    topicListAdapter = TopicListAdapter(itemList, fragment as TopicSummaryClickListener)
+
+    val homeLayoutManager = GridLayoutManager(activity.applicationContext, 2)
+    homeLayoutManager.spanSizeLookup = object : GridLayoutManager.SpanSizeLookup() {
+      override fun getSpanSize(position: Int): Int {
+        return if (position == 0 || position == 1) {
+          /* number of spaces this item should occupy = */ 2
+        } else {
+          /* number of spaces this item should occupy = */ 1
+        }
+      }
+    }
+
+    binding.homeRecyclerView.apply {
+      adapter = topicListAdapter
+      // https://stackoverflow.com/a/32763434/32763621
+      layoutManager = homeLayoutManager
+    }
     binding.let {
-      it.viewModel = getUserAppHistoryViewModel()
       it.presenter = this
       it.lifecycleOwner = fragment
     }
 
     userAppHistoryController.markUserOpenedApp()
 
+    subscribeToUserAppHistory()
+    subscribeToTopicList()
+
     return binding.root
   }
 
   private fun getUserAppHistoryViewModel(): UserAppHistoryViewModel {
     return viewModelProvider.getForFragment(fragment, UserAppHistoryViewModel::class.java)
   }
 
+  private fun getPromotedStoryViewModel(): PromotedStoryViewModel? {
+    return promotedStoryViewModel.getForFragment(fragment, PromotedStoryViewModel::class.java)
```

Comment:
Won't this always be non-null?

Reply:
@BenHenning I was thinking of the case when the learner has just started using the app, in that case what there won't be any promoted story, right? Or the domain layer is making sure that there is a promoted story every time?

Reply:
For now I am making this non-null based on your comment.